pkg/aws/manager_ec2_security_group.go

```diff
@@ -0,0 +1,114 @@
+package aws
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/integr8ly/cluster-service/pkg/clusterservice"
+	"github.com/integr8ly/cluster-service/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	loggingKeySecurityGroup = "security-group-id"
+
+	resourceTypeSecurtyGroup = "ec2:security-group"
+)
+
+var _ ClusterResourceManager = &SecurityGroupManager{}
+
+// SecurityGroupManager type
+type SecurityGroupManager struct {
+	ec2Client     ec2Client
+	taggingClient taggingClient
+	logger        *logrus.Entry
+}
+
+// NewDefaultSecurityGroupManager create session for manager
+func NewDefaultSecurityGroupManager(session *session.Session, logger *logrus.Entry) *SecurityGroupManager {
+	return &SecurityGroupManager{
+		ec2Client:     ec2.New(session),
+		taggingClient: resourcegroupstaggingapi.New(session),
+		logger:        logger.WithField(loggingKeyManager, managerSecurityGroup),
+	}
+}
+
+// GetName getter function
+func (r *SecurityGroupManager) GetName() string {
+	return "AWS EC2 SecurityGroup Manager"
+}
+
+// DeleteResourcesForCluster deletes resource for cluster
+func (r *SecurityGroupManager) DeleteResourcesForCluster(clusterId string, tags map[string]string, dryRun bool) ([]*clusterservice.ReportItem, error) {
+	var securityGroupsToDelete []*basicResource
```

Comment:
Is there a need to declare this slice here, when we can declare and return it in the `GetDeleteSecurityGroup` function?

Reply:
`GetDeleteSecurityGroup` doesn't need to exist, will pull that functionallity back to `DeleteResourcesForCluster` to match the pattern in the other managers.